fv3core/utils/corners.py

```diff
@@ -220,52 +221,116 @@ def definition(q: FloatField):
+        q = q[3, 2, 0]
+    with horizontal(region[i_start - 3, j_start - 2], region[i_start - 2, j_end + 3]):
+        q = q[4, -1, 0]
+    with horizontal(region[i_start - 2, j_start - 2], region[i_start - 2, j_end + 2]):
+        q = q[3, 0, 0]
+    with horizontal(region[i_start - 1, j_start - 2], region[i_start - 2, j_end + 1]):
+        q = q[2, 1, 0]
+    with horizontal(region[i_start - 3, j_start - 1], region[i_start - 1, j_end + 3]):
+        q = q[3, -2, 0]
+    with horizontal(region[i_start - 2, j_start - 1], region[i_start - 1, j_end + 2]):
+        q = q[2, -1, 0]
+    with horizontal(region[i_start - 1, j_start - 1], region[i_start - 1, j_end + 1]):
+        q = q[1, 0, 0]
+    with horizontal(region[i_end + 1, j_start - 3], region[i_end + 3, j_end + 1]):
+        q = q[-3, 2, 0]
+    with horizontal(region[i_end + 2, j_start - 3], region[i_end + 3, j_end + 2]):
+        q = q[-4, 1, 0]
+    with horizontal(region[i_end + 3, j_start - 3], region[i_end + 3, j_end + 3]):
+        q = q[-5, 0, 0]
+    with horizontal(region[i_end + 1, j_start - 2], region[i_end + 2, j_end + 1]):
+        q = q[-2, 1, 0]
+    with horizontal(region[i_end + 2, j_start - 2], region[i_end + 2, j_end + 2]):
+        q = q[-3, 0, 0]
+    with horizontal(region[i_end + 3, j_start - 2], region[i_end + 2, j_end + 3]):
+        q = q[-4, -1, 0]
+    with horizontal(region[i_end + 1, j_start - 1], region[i_end + 1, j_end + 1]):
+        q = q[-1, 0, 0]
+    with horizontal(region[i_end + 2, j_start - 1], region[i_end + 1, j_end + 2]):
+        q = q[-2, -1, 0]
+    with horizontal(region[i_end + 3, j_start - 1], region[i_end + 1, j_end + 3]):
+        q = q[-3, -2, 0]
+    return q
 
 
-def copy_ne_corner(q, direction, grid, kslice):
-    for j in range(grid.je + 1, grid.je + grid.halo + 1):
-        for i in range(grid.ie + 1, grid.ie + grid.halo + 1):
-            if direction == "x":
-                q[i, j, kslice] = q[j, 2 * (grid.ie + 1) - 1 - i, kslice]
-            if direction == "y":
-                q[i, j, kslice] = q[2 * (grid.je + 1) - 1 - j, i, kslice]
+@gtstencil
+def copy_corners_x_stencil(q: FloatField):
+    with computation(PARALLEL), interval(...):
+        q = copy_corners_x(q)
 
 
-def copy_nw_corner(q, direction, grid, kslice):
-    for j in range(grid.je + 1, grid.je + grid.halo + 1):
-        for i in range(grid.is_ - grid.halo, grid.is_):
-            if direction == "x":
-                q[i, j, kslice] = q[grid.je + 1 - j + 2, i - 2 + grid.ie, kslice]
-            if direction == "y":
-                q[i, j, kslice] = q[j + 2 - grid.ie, grid.je + 1 - i + 2, kslice]
+@gtstencil
+def copy_corners_y_stencil(q: FloatField):
+    with computation(PARALLEL), interval(...):
+        q = copy_corners_y(q)
 
 
-# can't actually be a stencil because offsets are variable
+# TODO: may want to put this in the test code if no longer used
+# directly in the model
```

Comment:
seems like with your refactor of the test, this is not used at all anymore?

Reply:
yes! I meant to delete this